moveit_ros/planning_interface/move_group_interface/include/moveit/move_group_interface/move_group_interface.h

```diff
@@ -177,15 +178,14 @@ class MoveGroupInterface
   MoveGroupInterface(const MoveGroupInterface&) = delete;
   MoveGroupInterface& operator=(const MoveGroupInterface&) = delete;
 
-  MoveGroupInterface(MoveGroupInterface&& other);
-  MoveGroupInterface& operator=(MoveGroupInterface&& other);
-
+  MoveGroupInterface(MoveGroupInterface&& other) noexcept;
+  MoveGroupInterface& operator=(MoveGroupInterface&& other) noexcept;
   /** \brief Get the name of the group this instance operates on */
   const std::string& getName() const;
 
   /** \brief Get the names of the named robot states available as targets, both either remembered states or default
    * states from srdf */
-  const std::vector<std::string> getNamedTargets();
+  std::vector<std::string> getNamedTargets() const;
```

Comment:
Thanks for pointing out the superfluous `const` here.

It seems this return value is not a const *reference* because the authors wanted to avoid raising an exception if the JointModelGroup does not exist. Instead they return a new empty vector (which mustn't be passed by reference).

However, [the constructor of the class guarantees](https://github.com/ros-planning/moveit/blob/master/moveit_ros/planning_interface/move_group_interface/src/move_group_interface.cpp#L107) the group exists in this robot model.

So instead of removing const-ness of the return value, please turn it into a const-reference and just remove the internal logic that can syntactically return the new empty vector.

Reply:
Please also expand the comment to explain briefly why its this way, to avoid future ppl from being confused

Reply:
It really seems that there is a guarantee about `jointModelGroup` always existing. Will change they way you suggested.